src/mod_muc.erl

```diff
@@ -809,25 +809,30 @@ iq_disco_info(Lang) ->
 
 
 -spec iq_disco_items(jid:server(), jid:jid(), ejabberd:lang(),
-        Rsm :: none | jlib:rsm_in()) -> any().
+                     Rsm :: none | jlib:rsm_in()) -> any().
 iq_disco_items(Host, From, Lang, none) ->
-    lists:zf(fun(#muc_online_room{name_host = {Name, _Host}, pid = Pid}) ->
+    Online = lists:zf(fun(#muc_online_room{name_host = {Name, _Host}, pid = Pid}) ->
                      case catch gen_fsm_compat:sync_send_all_state_event(
                                   Pid, {get_disco_item, From, Lang}, 100) of
-                         {item, Desc} ->
+                         {item, _Desc} ->
```

Comment:
It breaks the XEP, because the `name` attribute must be equal to room configured name, not username from JID. https://xmpp.org/extensions/xep-0045.html#disco-rooms